api/leetcode-data.js

```diff
@@ -0,0 +1,104 @@
+import { dailyGetLeetcodeData, weeklyGetLeetcodeData, getProblemSet } from "./leetcode/index.js";
+import logger from "../logging.js";
+import staticProblemSets from './leetcode/static/problemSets.json' assert { type: "json" };
+
+export class LeetcodeData {
+
+    constructor() {}
```

Comment:
nit: is there any reason you want to have this empty constructor? I usually don't have them around unless there's some important initialization.